plugins/grafana-custom-plugins/grafana-dependency-plugin/src/DependencyPanel.tsx

```diff
@@ -47,124 +53,162 @@ export const DependencyPanel: React.FC<Props> = ({ options, data, width, height
       break;
   }
 
-  mermaid.initialize({
-    startOnLoad: true,
-    theme: 'base',
-    themeVariables: {
-      primaryColor: boxColor,
-      secondaryColor: theme.colors.background.canvas,
-      tertiaryColor: theme.colors.background.canvas,
-      primaryTextColor: theme.colors.text.maxContrast,
-      lineColor: theme.colors.text.maxContrast,
-    },
-  });
-
-  for (let i = 0; i < frame.length; i++) {
-    const sourcePodName = sourcePodNames?.values.get(i);
-    const sourcePodLabel = sourcePodLabels?.values.get(i);
-    const sourceNodeName = sourceNodeNames?.values.get(i);
-    const destinationPodName = destinationPodNames?.values.get(i);
-    const destinationPodLabel = destinationPodLabels?.values.get(i);
-    const destinationNodeName = destinationNodeNames?.values.get(i);
-    const destinationServicePortName = destinationServicePortNames?.values.get(i);
-    const octetDeltaCount = octetDeltaCounts?.values.get(i);
-
-    function getName(groupByLabel: boolean, source: boolean, labelJSON: string) {
-      if(!groupByLabel || labelJSON === undefined || options.labelName === undefined) {
-        return source ? sourcePodName : destinationPodName;
+  function layerFourGraph() {
+    for (let i = 0; i < frame.length; i++) {
+      const sourcePodName = sourcePodNames?.values.get(i);
+      const sourcePodLabel = sourcePodLabels?.values.get(i);
+      const sourceNodeName = sourceNodeNames?.values.get(i);
+      const destinationPodName = destinationPodNames?.values.get(i);
+      const destinationPodLabel = destinationPodLabels?.values.get(i);
+      const destinationNodeName = destinationNodeNames?.values.get(i);
+      const destinationServicePortName = destinationServicePortNames?.values.get(i);
+      const octetDeltaCount = octetDeltaCounts?.values.get(i);
+  
+      function getName(groupByLabel: boolean, source: boolean, labelJSON: string) {
+        if(!groupByLabel || labelJSON === undefined || options.labelName === undefined) {
+          return source ? sourcePodName : destinationPodName;
+        }
+        let labels = JSON.parse(labelJSON);
+        if(labels[options.labelName] !== undefined) {
+          return labels[options.labelName];
+        }
+        return sourcePodName;
+      }
+  
+      let groupByPodLabel = options.groupByPodLabel;
```

Comment:
same question, can we reuse `options.groupByPodLabel` without introducing new variable?

Reply:
yes, i can change this one